packages/cli/test/unit/decorators/registerController.test.ts

```diff
@@ -4,36 +4,112 @@ jest.mock('@/constants', () => ({
-		endpoint() {
-			return { ok: true };
+	const license = mock<License>();
+	const authService = mock<AuthService>();
+	let agent: SuperAgentTest;
+
+	beforeEach(() => {
+		jest.resetAllMocks();
+		const app = express();
+		new ControllerRegistry(license, authService).activate(app);
+		agent = testAgent(app);
+	});
+
+	describe('Rate limiting', () => {
+		@RestController('/test')
+		// @ts-expect-error tsc complains about unused class
+		class TestController {
+			@Get('/unlimited')
+			unlimited() {
+				return { ok: true };
+			}
+
+			@Get('/rate-limited', { rateLimit: {} })
+			rateLimited() {
+				return { ok: true };
+			}
 		}
-	}
 
-	mockInstance(AuthService);
-	const app = express();
-	registerController(app, TestController);
-	const agent = testAgent(app);
+		beforeEach(() => {
+			authService.authMiddleware.mockImplementation(async (_req, _res, next) => next());
+		});
+
+		it('should not rate-limit by default', async () => {
+			for (let i = 0; i < 6; i++) {
+				await agent.get('/rest/test/unlimited').expect(200);
+			}
+		});
+
+		it('should rate-limit when configured', async () => {
+			for (let i = 0; i < 5; i++) {
+				await agent.get('/rest/test/rate-limited').expect(200);
+			}
+			await agent.get('/rest/test/rate-limited').expect(429);
+		});
+	});
 
-	it('should not rate-limit by default', async () => {
-		for (let i = 0; i < 6; i++) {
-			await agent.get('/rest/test/unlimited').expect(200);
+	describe('Authorization', () => {
+		@RestController('/test')
+		// @ts-expect-error tsc complains about unused class
+		class TestController {
+			@Get('/no-auth', { skipAuth: true })
+			noAuth() {
+				return { ok: true };
+			}
+
+			@Get('/auth')
+			auth() {
+				return { ok: true };
+			}
 		}
+
+		it('should not require auth unless configured', async () => {
+			await agent.get('/rest/test/no-auth').expect(200);
+			expect(authService.authMiddleware).not.toHaveBeenCalled();
+		});
+
+		it('should require auth by default', async () => {
+			authService.authMiddleware.mockImplementation(async (_req, res) => {
+				res.status(401).send();
+			});
+			await agent.get('/rest/test/auth').expect(401);
+			expect(authService.authMiddleware).toHaveBeenCalled();
+		});
 	});
 
-	it('should rate-limit when configured', async () => {
-		for (let i = 0; i < 5; i++) {
-			await agent.get('/rest/test/rate-limited').expect(200);
+	describe('License checks', () => {
+		@RestController('/test')
+		// @ts-expect-error tsc complains about unused class
+		class TestController {
+			@Get('/with-sharing')
+			@Licensed('feat:sharing')
+			sharing() {
+				return { ok: true };
+			}
 		}
-		await agent.get('/rest/test/rate-limited').expect(429);
+
+		beforeEach(() => {
+			authService.authMiddleware.mockImplementation(async (_req, _res, next) => next());
+		});
+
+		it('should not allow when feature is not available', async () => {
+			license.isFeatureEnabled.calledWith('feat:sharing').mockReturnValue(false);
```

Comment:
I think Val created `LicenseMocker` to simplify this, if it can be used here.

Reply:
chose to keep using `mock<License>()` in here for now because that class seems to be created for integration tests, and this is more of a unit test.